torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py

```diff
@@ -652,14 +652,15 @@ def extra_repr(self) -> str:
             f"sharding_groups={self.sharding_groups}")
     return repr
 
-  def __getattr__(self, name: str) -> Any:
+  def __getattr__(self, name: str) -> Union[Tensor, nn.Module]:
     """Forward missing attributes to wrapped module."""
     try:
       return super().__getattr__(name)  # defer to nn.Module's logic
     except AttributeError:
       return getattr(self.module, name)
 
-  def __getitem__(self, key: int) -> Any:
+  T = TypeVar('T', bound=Module)
+  def __getitem__(self, key: int) -> Union[nn.Sequential, T]:
```

Comment:
same here, I think we can just use `def __getitem__(self, key: int) -> nn.Module:` as the signature here.

Reply:
Changed